### PR TITLE
[dtensor] switch aten.t to use op strategy

### PR DESCRIPTION
**Pull request by wanchaol**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #122950
* #122949
* #122929

as titled

cc @mrshenli @pritamdamania87 @zhaojuanmao @satgera @rohan-varma @gqchen @aazzolini @osalpekar @jiayisuse @H-Huang @kwen2501 @awgu @penguinwu @fegin @XilunWu @fduwjj @wz337 @tianyu-l @wconstab @yf225 @chauhang

### CODE REVIEW

File: torch/distributed/_tensor/ops/matrix_ops.py [hunk 1]
```diff
@@ -32,9 +25,29 @@
...
+def transpose_strategy(mesh: DeviceMesh, op_schema: OpSchema) -> OpStrategy:
+    self_strategy = op_schema.args_schema[0]
+    assert isinstance(self_strategy, OpStrategy)
+
+    transpose_strategies = []
+    for input_strategy in self_strategy.strategies:
+        input_spec = input_strategy.output_spec
+        # follow the input spec but transpose the Shard placements
+        output_placements = [
+            Shard(1 - p.dim) if isinstance(p, Shard) else p
```
Comment (awgu) [inline torch/distributed/_tensor/ops/matrix_ops.py:38]: Does this assume 2D mesh?

Comment (wanchaol) [reply torch/distributed/_tensor/ops/matrix_ops.py:38]: I don't think it does, it should work for n-D mesh

basically the reason is that `aten.t` can only be called when your input tensor is 2 dimensional (ndim == 2), so the transpose happen locally can change the sharding dimension from either `0->1` or `1 -> 0`

Comment (awgu) [reply torch/distributed/_tensor/ops/matrix_ops.py:38]: This makes sense! I did not realize before that `aten.t` only works for 2D tensor.

**Review by awgu — APPROVED** [review]
LGTM!

**Review by tianyu-l — APPROVED** [review]
lgtm!

**Pull request closed without merging**